ci/scala.yml

```diff
@@ -6,17 +11,24 @@ on:
   pull_request:
     branches: [ $default-branch ]
 
+permissions:
+  contents: read
+
 jobs:
   build:
 
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up JDK 11
-      uses: actions/setup-java@v2
+      uses: actions/setup-java@v3
       with:
         java-version: '11'
-        distribution: 'adopt'
+        distribution: 'temurin'
+        cache: 'sbt'
     - name: Run tests
       run: sbt test
+      # This step uploads information to the GitHub dependency graph and unblocking Dependabot alerts for the repository
+    - name: Upload dependency graph
+      uses: scalacenter/sbt-dependency-submission@v2
```

Comment:
Can we use commit sha instead? Latest one seems this: `ab086b50c947c9774b70f39fc7f6e20ca2706c91`

Reply:
Is this a mandatory step or optional?

Reply:
Fixed both of those.